examples/daal/cpp/source/k_nearest_neighbors/bf_knn_dense_search_batch.cpp

```diff
@@ -0,0 +1,194 @@
+
+/* Input data set parameters */
+const string trainDatasetFileName =
+    "../data/batch/k_nearest_neighbors_train.csv";
+const string testDatasetFileName = "../data/batch/k_nearest_neighbors_test.csv";
+const string groundTruthDistancesFileName =
+    "../data/batch/k_nearest_neighbors_distances_ground_truth.csv";
+const string groundTruthIndicesFileName =
+    "../data/batch/k_nearest_neighbors_indices_ground_truth.csv";
+
+const size_t nFeatures = 5;
+const size_t nClasses = 5;
+const size_t kNeighbors = 3;
+
+void trainModel(bf_knn_classification::training::ResultPtr &trainingResult);
+void testModel(bf_knn_classification::training::ResultPtr &trainingResult,
+               bf_knn_classification::prediction::ResultPtr &predictionResult);
+void readGroundTruth(NumericTablePtr &groundTruthIndices,
+                     NumericTablePtr &groundTruthDistances);
+void printIndicesResults(
+    NumericTablePtr &testGroundTruth,
+    bf_knn_classification::prediction::ResultPtr &predictionResult);
+void printDistancesResults(
+    NumericTablePtr &testGroundTruth,
+    bf_knn_classification::prediction::ResultPtr &predictionResult);
+
+int main(int argc, char *argv[]) {
+  checkArguments(argc, argv, 4, &trainDatasetFileName, &testDatasetFileName,
+                 &groundTruthDistancesFileName, &groundTruthIndicesFileName);
+
+  bf_knn_classification::training::ResultPtr trainingResult;
+  trainModel(trainingResult);
+
+  bf_knn_classification::prediction::ResultPtr searchResult;
+  testModel(trainingResult, searchResult);
+
+  NumericTablePtr groundTruthDistances, groundTruthIndices;
+  readGroundTruth(groundTruthIndices, groundTruthDistances);
+
+  printIndicesResults(groundTruthIndices, searchResult);
+  printDistancesResults(groundTruthDistances, searchResult);
+
+  return 0;
+}
+
+void trainModel(bf_knn_classification::training::ResultPtr &trainingResult) {
+  /* Initialize FileDataSource<CSVFeatureManager> to retrieve the input data
+   * from a .csv file */
+  FileDataSource<CSVFeatureManager> trainDataSource(
+      trainDatasetFileName, DataSource::notAllocateNumericTable,
+      DataSource::doDictionaryFromContext);
+
+  /* Create Numeric Tables for training data and labels */
+  NumericTablePtr trainData =
+      HomogenNumericTable<>::create(nFeatures, 0, NumericTable::notAllocate);
+  NumericTablePtr trainGroundTruth =
+      HomogenNumericTable<>::create(1, 0, NumericTable::doNotAllocate);
+  NumericTablePtr mergedData(
+      new MergedNumericTable(trainData, trainGroundTruth));
+  /* Retrieve the data from the input file */
+  trainDataSource.loadDataBlock(mergedData.get());
+
+  /* Create an algorithm object to train the BF kNN model */
+  bf_knn_classification::training::Batch<> algorithm;
+
+  /* Pass the training data set and dependent values to the algorithm */
+  algorithm.input.set(classifier::training::data, trainData);
+  algorithm.input.set(classifier::training::labels, trainGroundTruth);
```

Comment:
For search labels not needed?

Reply:
1) Consistency
2) To show that both things can be done simultaneously

Reply:
Consistency with what? This is how we recommend using the algorithm. Where does the user get the labels for the search task?

Reply:
Example should be minimal. If you want to show both, why we need 2 examples instead of 1?